mmv1/third_party/terraform/services/resourcemanager/fw_data_source_google_project.go

```diff
@@ -0,0 +1,158 @@
+package resourcemanager
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	"github.com/hashicorp/terraform-provider-google/google/fwmodels"
+	"github.com/hashicorp/terraform-provider-google/google/fwtransport"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+	"google.golang.org/api/cloudresourcemanager/v1"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &GoogleProjectFWDataSource{}
+	_ datasource.DataSourceWithConfigure = &GoogleProjectFWDataSource{}
+)
+
+// NewGoogleProjectDataSource is a helper function to simplify the provider implementation.
+func NewGoogleProjectFWDataSource() datasource.DataSource {
+	return &GoogleProjectFWDataSource{}
+}
+
+// GoogleProjectDataSource is the data source implementation.
+type GoogleProjectFWDataSource struct {
+	client         *cloudresourcemanager.Service
+	providerConfig *transport_tpg.Config
+}
+
+type GoogleProjectModel struct {
+	Id        types.String `tfsdk:"id"`
+	ProjectId types.String `tfsdk:"project_id"`
+	Name      types.String `tfsdk:"name"`
+	OrgId     types.String `tfsdk:"org_id"`
+	FolderId  types.String `tfsdk:"folder_id"`
+	Number    types.String `tfsdk:"number"`
+}
+
+// Metadata returns the data source type name.
+func (d *GoogleProjectFWDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_fwprovider_project"
```

Comment:
shall we shorten to something like `_fw_project` or `project_fw`? We should probably standardize to something for now. Sarah had https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/fwprovider/data_source_provider_config_plugin_framework.go#L58, but that feels different enough